files/bootstrap.sh

```diff
@@ -112,6 +112,77 @@ function get_pause_container_account_for_region () {
     esac
 }
 
+# Gets resource to reserve in a given region
+# Args:
+#   $1 total available resource in input unit
+#   $2 start of region in input unit
+#   $3 end of region in input unit
+#   $4 percentage of region to reserve in percent*100 (to allow for two decimal digits)
+#   $5 multiplier from input unit to output unit
+# Output: resource to reserve in output unit
+get_resource_to_reserve_in_region() {
+  local total=$1
+  local start=$2
+  local end=$3
+  local percentage=$4
+  local multiplier=$5
+  resource="0"
+  if (( $total > $start )); then
+    resource=$(((($total < $end ? $total : $end) - $start) * $multiplier * $percentage / 100 / 100))
+  fi
+  echo $resource
+}
+
+# Gets memory to reserve in mebibytes
+# Args:
+#   $1 total available memory in Gi
+# Output: memory to reserve in Mi
+get_memory_mebibytes_to_reserve() {
+  local total=$1
+  local multiplier=1024
+  # 0   4   8   16  128  $total
+  #  25% 20% 10%  6%   2%
+  local regions=(0 4 8 16 128 $total)
+  local percentages=(2500 2000 1000 600 200)
+  if (( $total <= 1 )); then
```

Comment:
This won't work for machines with less than 1 GB of memory, because bash doesn't support math for non-integers AFAIK.  Out of curiosity, why did you pick bash and not python, since theres a decent amount of calculation here?  I'd prefer the latter, and I think the math would be easier to follow and easier to test.  

(There *are* machines with less than 1GB of memory, for example the [t3.nano)](https://aws.amazon.com/ec2/instance-types/).  Whether or not it's realistic for someone to use one, we shouldn't break if they find a way.

Reply:
python is not installed.

We could install it then run the script then uninstall it all during bootstrap.

I think that would take an unacceptable amount of time though

Reply:
Also i think it still "works' for machines less than 1GB since if $total is 0, 256 MiB is returned. Anyway, switching from GB to MiB will make this a non-issue.

Reply:
Actually python2 is installed 🤦‍♂ I never even checked.

Let's leave it as a TODO to convert to a python script. Long-term, this much bash will be hard to maintain.